client/state/site-settings/reducer.js

```diff
@@ -64,7 +66,21 @@ export const items = createReducer( {}, {
 			...state[ siteId ],
 			...settings
 		}
-	} )
+	} ),
+	[ MEDIA_DELETE ]: ( state, { siteId, mediaIds } ) => {
```

Comment:
Hmm, given my note at https://github.com/Automattic/wp-calypso/pull/10016#issuecomment-269004720, maybe we don't even need to handle this.